Snippets/Core/Core_6/Handlers/AsyncHandler.cs

```diff
@@ -0,0 +1,190 @@
+﻿namespace Core6.Handlers
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Common;
+    using NServiceBus;
+
+    #region ShortComputeBoundMessageHandler
+    public class ShortComputeBoundHandler : IHandleMessages<MyMessage>
+    {
+        public Task Handle(MyMessage message, IMessageHandlerContext context)
+        {
+            ComputeBoundComponent.BlocksForAFewMilliseconds();
+            return Task.FromResult(0); // or Task.CompletedTask
+        }
+    }
+    #endregion
+
+    #region LongComputeBoundMessageHandler
+    public class LongComputeBoundHandler : IHandleMessages<MyMessage>
+    {
+        public Task Handle(MyMessage message, IMessageHandlerContext context)
+        {
+            return Task.Run(() => ComputeBoundComponent.BlocksForMoreThanHundredMilliseconds());
+        }
+    }
+    #endregion
+
+    public class ComputeBoundComponent
+    {
+        public static void BlocksForAFewMilliseconds()
+        {
+        }
+
+        public static void BlocksForMoreThanHundredMilliseconds()
+        {
+        }
+    }
+
+    #region HandlerReturnsATask
+    public class HandlerReturnsATask : IHandleMessages<MyMessage>
+    {
+        public Task Handle(MyMessage message, IMessageHandlerContext context)
+        {
+            return SomeLibrary.SomeAsyncMethod(message);
```

Comment:
It's probably clear enough but you could make this super-explicit with a temp variable:

```CSharp
var task = SomeLibrary.SomeAsyncMethod(message);
return task;
```